networks/p2p/dial.go

```diff
@@ -464,7 +466,11 @@ func (s *dialstate) checkDial(n *discover.Node, peers map[discover.NodeID]*Peer)
 func (s *dialstate) taskDone(t task, now time.Time) {
 	switch t := t.(type) {
 	case *dialTask:
-		s.hist.add(t.dest.ID, now.Add(dialHistoryExpiration))
+		// Updated dial can be redialed right away
+		if !t.updating {
+			s.hist.add(t.dest.ID, now.Add(dialHistoryExpiration))
```

Comment:
If we add `updating` variable, do we need to reset the flag for the second try of multi channel?
```suggestion
			s.hist.add(t.dest.ID, now.Add(dialHistoryExpiration))
			t.updating == false
```

Reply:
It' more complicated... Please find good solution. :)

Reply:
does this look better?
```
		delete(s.dialing, t.dest.ID)
		if it.updating {
			// reset value for next dial
			// otherwise, redial can happen every second
			t.updating = false
			// Updated dial can be redialed right away
			return
		}
		s.hist.add(t.dest.ID, now.Add(dialHistoryExpiration))		
```